### PR TITLE
add optimization for blip text model generation

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sywangyi** [comment]
@regisss

**Comment by sywangyi** [comment]
@libinta @mandy-li please have review

**Comment by sywangyi** [comment]
two tasks could benefit from the optimization.
image-to-text. models are like "Salesforce/blip-image-captioning-large" and "Salesforce/blip-image-captioning-base"
visual-question-answering. models are like "Salesforce/blip-vqa-base" and "Salesforce/blip-vqa-capfilt-large". although VQA has another issue of torch.full type. filed in https://habana.atlassian.net/servicedesk/customer/portal/1/HS-1395.
the acceleration ratio in Gaudi2：
image to text: Salesforce/blip-image-captioning-large. generation speedup:238ms vs 40ms(after)
                       Salesforce/blip-image-captioning-base: generation speedup: 227ms vs 25ms(after)

visual-question-answering. 
                      Salesforce/blip-vqa-base: generation speedup: 86ms vs 9ms(after)
                       Salesforce/blip-vqa-capfilt-large:generation speedup: 91ms vs 10ms(after)

**Comment by sywangyi** [comment]
test codes are like
from transformers import pipeline
import habana_frameworks.torch
import torch
import requests
import PIL.Image
from habana_frameworks.torch.hpu import wrap_in_hpu_graph
import time
from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi
adapt_transformers_to_gaudi()


image_url = "https://ankur3107.github.io/assets/images/image-captioning-example.png"
image = PIL.Image.open(requests.get(image_url, stream=True, timeout=3000).raw)


generator = pipeline(
        "image-to-text",
        model="Salesforce/blip-image-captioning-large",
        torch_dtype=torch.bfloat16,
        device="hpu",
)
generator.model.text_decoder = wrap_in_hpu_graph(generator.model.text_decoder)
generator.model.vision_model = wrap_in_hpu_graph(generator.model.vision_model)
for i in range(5):
    generator(image)


start = time.time()
result = generator(image)
end = time.time()
print(f"{result}, time = {(end-start) * 1000}")

**Comment by sywangyi** [comment]
upload a PR to fix the issue in  https://habana.atlassian.net/servicedesk/customer/portal/1/HS-1395.

**Comment by sywangyi** [comment]
I check with latest transformers 4.37.2. this PR also work. could somebody review the PR?

**Comment by sywangyi** [comment]
from transformers import pipeline
import habana_frameworks.torch
import torch
import requests
import PIL.Image
from habana_frameworks.torch.hpu import wrap_in_hpu_graph
import time
from optimum.habana.transformers.modeling_utils import adapt_transformers_to_gaudi
adapt_transformers_to_gaudi()



image = PIL.Image.open("./vqa_cats.jpg").convert('RGB')

generator = pipeline(
"visual-question-answering",
model="Salesforce/blip-vqa-capfilt-large",
torch_dtype=torch.bfloat16,
device="hpu",
)
for i in range(5):
   generator(image, "how many dogs are in the picture?", topk=1, max_new_tokens=20)

start = time.time()
result = generator(image, "how many dogs are in the picture?", topk=1, max_new_tokens=20)
end = time.time()
print(f"{result}, time = {(end-start) * 1000}")

**Review by regisss — COMMENTED** [review]
I left a couple of minor comments.
Also, could you add this model to the table [here](https://github.com/huggingface/optimum-habana?tab=readme-ov-file#validated-models) and [there](https://github.com/huggingface/optimum-habana/blob/main/docs/source/index.mdx) please?

**Review by regisss — COMMENTED** [review]
Can you run the following from the root of the repo please?
```
pip install -U ruff
make style
```

Also, do you have any insight into the issue with static shapes?

File: optimum/habana/transformers/models/blip/modeling_blip.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        - torch.full add dtype=torch.int64, or else the default type is torch.float32. lead to coredump in embeding layer
+    """
+    from habana_frameworks.torch.hpu import wrap_in_hpu_graph
+
+    if not hasattr(self.vision_model, "clear_cache"):
+        self.vision_model = wrap_in_hpu_graph(self.vision_model)
+    if not hasattr(self.text_encoder, "clear_cache"):
+        self.text_encoder = wrap_in_hpu_graph(self.text_encoder)
+    if not hasattr(self.text_decoder, "clear_cache"):
+        self.text_decoder = wrap_in_hpu_graph(self.text_decoder)
```
[the remark this answers is not in the document]
Comment (sywangyi) [reply optimum/habana/transformers/models/blip/modeling_blip.py:83]: for visual-question-answering pipeline, no generation_kwargs will be passed to generate, how to handle this case here? see https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/visual_question_answering.py#L128

**Comment by sywangyi** [comment]
> I left a couple of minor comments. Also, could you add this model to the table [here](https://github.com/huggingface/optimum-habana?tab=readme-ov-file#validated-models) and [there](https://github.com/huggingface/optimum-habana/blob/main/docs/source/index.mdx) please?

after this PR is merged, I need to upload example for VQA and image-to-text case. currently there's no such example in optimum-habana, then update doc to add the model and example

File: optimum/habana/transformers/models/blip/modeling_blip.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sywangyi) [reply optimum/habana/transformers/models/blip/modeling_blip.py]: yes, we could use the default behavior is to use HPU graph now

**Review by regisss — APPROVED** [review]
LGTM

**Pull request merged**